osu.Game.Tests/Beatmaps/Formats/LegacyBeatmapEncoderTest.cs

```diff
@@ -76,6 +77,37 @@ public void TestEncodeMultiSegmentSliderWithFloatingPointError()
             Assert.That(decodedSlider.Path.ControlPoints.Count, Is.EqualTo(5));
         }
 
+        [TestCaseSource(nameof(allBeatmaps))]
+        public void TestEncodeDecodeStabilityWithNonLegacyControlPoints(string name)
+        {
+            var decoded = decodeFromLegacy(beatmaps_resource_store.GetStream(name), name);
+
+            sort(decoded.beatmap);
+
+            var originalSerialized = decoded.beatmap.Serialize();
+            var encoded = encodeToLegacy(decoded);
```

Comment:
I'm confused as to why this is encoded here exactly and not later.

* The map is decoded to the `decoded` variable from resources in line 83.
* Then it is serialised to `originalSerialized` in line 87 for later comparison.
* Here, in line 88, it is encoded to the `encoded` variable again.
* Lines 90 through 101 mutate `decoded`.
* In line 103 `encoded` is decoded again, and then is sorted, serialised and then compared to `originalSerialized`. But `encoded` was actually written *before* the mutation in lines 90-101, so isn't this just testing normal stability...?

Mind you the test still passes after this encode call is moved immediately before line 103, I'm just confused as to what is being tested in the version as written.

Reply:
I think the logic changed and it will work now as you state, but I moved the encoding to before mutating the control points as it used to fail i otherwise (the json would not match).

Reply:
Revisiting this as I reached the same conclusion (there's no way this test should be passing as it is), your thinking is correct, and this isn't correctly testing.